tools/scripts/stm32.mk

```diff
@@ -60,11 +88,25 @@ ITC = $(call rwildcard, $(PROJECT_BUILD)/Src,*_it.c)
 HALCONF = $(call rwildcard, $(PROJECT_BUILD)/Inc,*_hal_conf.h)
 
 ifneq (,$(wildcard $(PROJECT_BUILD)))
+
+ifeq ($(STM32_EXT_BUILD),y)
+
+TARGET = $(shell echo $(CHIPNAME) | grep -Eo 'STM32[A-Z][0-9][0-9A-Z][0-9]')
```

Comment:
why duplicate the pattern matching below ? this will need lots of effort to maintain